### ISSUE TITLE
//tensorflow/core:platform_profile_utils_cpu_utils_test is failing on ppc64le with "Check failed: cpu_frequency > 0 (-1 vs. 0)"

### ISSUE THREAD

**Issue by sandipmgiri** [tensorflow/tensorflow#10450]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
    Ubuntu 16.04 (ppc64le)
- **TensorFlow installed from (source or binary)**:
    Installed from source (v1.0.1)
- **TensorFlow version (use command below)**:
   ('v1.0.1-0-ge895d5c-dirty', '1.0.1')
- **Bazel version (if compiling from source)**:
    0.4.4-2017-05-26 (@80a07b5)
- **CUDA/cuDNN version**:
    CUDA = 8.0 and cuDNN = 5.1
- **GPU model and memory**:
    GPU 0: Tesla P100-SXM2-16GB
    GPU 1: Tesla P100-SXM2-16GB
- **Exact command to reproduce**:
   bazel test --config=opt --config=cuda  //tensorflow/core:platform_profile_utils_cpu_utils_test

### Describe the problem
This test is passing successfully on X86, however getting failure on ppc64le with following errors:

### Source code / logs
```
$ bazel test --config=opt --config=cuda  //tensorflow/core:platform_profile_utils_cpu_utils_test

exec ${PAGER:-/usr/bin/less} "$0" || exit 1
-----------------------------------------------------------------------------
Running main() from test_main.cc
[==========] Running 5 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 5 tests from CpuUtilsTest
[ RUN      ] CpuUtilsTest.SetUpTestCase
[       OK ] CpuUtilsTest.SetUpTestCase (0 ms)
[ RUN      ] CpuUtilsTest.TearDownTestCase
[       OK ] CpuUtilsTest.TearDownTestCase (0 ms)
[ RUN      ] CpuUtilsTest.CheckGetCurrentClockCycle
[       OK ] CpuUtilsTest.CheckGetCurrentClockCycle (0 ms)
[ RUN      ] CpuUtilsTest.CheckCycleCounterFrequency
F tensorflow/core/platform/profile_utils/cpu_utils_test.cc:56] Check failed: cpu_frequency > 0 (-1 vs. 0)
external/bazel_tools/tools/test/test-setup.sh: line 123:  7285 Aborted                 (core dumped) "${TEST_PATH}" "$@"
```

Any comments/suggestions ?

**Comment by michaelisard** [comment]
What is the output of
`grep '^bogomips' /proc/cpuinfo | head -1`

**Comment by sandipmgiri** [comment]
Getting something on X86 : 
```
$ grep '^bogomips' /proc/cpuinfo | head -1
bogomips        : 4799.99
```
However nothing printed on ppc64le : 

`$ grep '^bogomips' /proc/cpuinfo | head -1`

**Comment by michaelisard** [comment]
Unfortunately we don't directly support the PPC platform, so there may be a bunch of issues like this. Marking community support in case somebody is willing to take on the attempt of getting it to work on that platform.

**Comment by sandipmgiri** [comment]
I gone through the file tensorflow/core/platform/profile_utils/cpu_utils_test.cc and found that GetCycleCounterFrequency function returning incorrect result on ppc64le,  expected value is greater than "0", however on ppc64le getting "-1"  (see below for test code)

```
TEST_F(CpuUtilsTest, CheckCycleCounterFrequency) {
  const int64 cpu_frequency = CpuUtils::GetCycleCounterFrequency();
  //printf("CPU FREQUENCY %d\n",cpu_frequency);
  CHECK_GT(cpu_frequency, 0);
  CHECK_NE(cpu_frequency, CpuUtils::INVALID_FREQUENCY);
  if (DBG) {
    LOG(INFO) << "Cpu frequency = " << cpu_frequency;
  }
}

```

Checked the output of CpuUtils::GetCycleCounterFrequency() using print statement i.e. `const int64 cpu_frequency = CpuUtils::GetCycleCounterFrequency();`

On x86 `cpu_frequency` = 239999500
On ppc64le `cpu_frequency` = -1

I would like to explore more on this. Please provide comments/suggestions if any. Thanks!

**Comment by sandipmgiri** [comment]
After investigation found that, "int64 cpu_frequency" value on x86 is "unsigned" by default whereas on ppc64le it is "signed" by default, that is why this test was failing on ppc64le.

To make the results compatible with x86, some code changes were required on ppc64le.
After code changes, I ran this test and now it is passing on ppc64le as well as on X86.

Code changes submitted for review in PR https://github.com/tensorflow/tensorflow/pull/10522  to fix this.
Once approved I will push changes to Github so PR https://github.com/tensorflow/tensorflow/pull/10522 should resolve this (relevant commit  - https://github.com/tensorflow/tensorflow/commit/c63cebb80a26cb7eeaf0afe095886488f1a8ac8a).

**Comment by sandipmgiri** [comment]
Community accepted the changes and merged in master (relevant PR #10522 and merged commit d0d2308). Hence closing this this as resolved.
